lib/lib-storage/src/s3-transfer-manager/transfer-manager-types.ts

```diff
@@ -0,0 +1,331 @@
+   *
+   * @param options.bucket - The name of the bucket
+   * @param options.destination - The destination directory
+   * @param options.s3Prefix - Specify the S3 prefix that limits the response to keys that begin with the specified prefix
+   * @param options.s3Delimiter - Specify the S3 delimiter.
+   * @param options.recursive - Whether to upload directories recursively.
+   * @param options.filter - A callback to allow users to filter out unwanted S3 object. It is invoked for each S3 object. An example implementation is a predicate that takes an S3Object and returns a boolean indicating whether this S3Object should be downloaded
+   * @param options.getObjectRequestCallback - A callback mechanism to allow customers to update individual getObjectRequest that the S3 Transfer Manager generates.
+   * @param options.failurePolicy - The failure policy to handle failed requests
+   * @param eventListeners - Collection of callbacks for monitoring transfer lifecycle events. Allows tracking statuses of directory downloads.
+   *
+   * @returns The number of objects that have been uploaded and the number of objects that have failed
+   */
+  downloadAll(options: {
+    bucket: string;
+    destination: string;
+    s3Prefix?: string;
+    s3Delimiter?: string;
+    recursive?: boolean;
+    filter?: (object?: S3Object) => boolean;
+    getObjectRequestCallback?: any;
+    failurePolicy?: any;
+    eventListeners?: TransferEventListeners;
+  }): Promise<{
+    objectsDownloaded: Long;
+    objectsFailed: Long;
+  }>;
+
+  /**
+   * Registers a callback function to be executed when a specific transfer event occurs.
+   * Supports monitoring the full lifecycle of transfers.
+   *
+   * @param type - The type of event to listen for.
+   * @param callback - Function to execute when the specified event occurs.
+   * @param options - Optional configuration for the event listener.
+   *
+   * @public
+   */
+  addEventListener(
+    type: "transferInitiated",
+    callback: EventHandler<TransferEvent>,
+    options?: AddEventListenerOptions | boolean
+  ): void;
+  addEventListener(
+    type: "bytesTransferred",
+    callback: EventHandler<TransferEvent>,
+    options?: AddEventListenerOptions | boolean
+  ): void;
+  addEventListener(
+    type: "transferComplete",
+    callback: EventHandler<TransferCompleteEvent>,
+    options?: AddEventListenerOptions | boolean
+  ): void;
+  addEventListener(
+    type: "transferFailed",
+    callback: EventHandler<TransferEvent>,
+    options?: AddEventListenerOptions | boolean
+  ): void;
+  addEventListener(type: string, callback: EventHandler | null, options?: AddEventListenerOptions | boolean): void;
+
+  /**
+   * Dispatches an event to the registered event listeners.
+   * Triggers callbacks registered via addEventListener with matching event types.
+   *
+   * @param event - The event object to dispatch.
+   * @returns True if event was successfuly dispatched, false otherwise.
```

Comment:
for boolean descriptions, use "whether" and only describe one case. The alternative is always implied. 

```
@returns whether the event dispatched successfully
```